node/src/main.rs

```diff
@@ -598,133 +632,358 @@ async fn main() -> Result<(), Box<dyn Error>> {
-                                         let status = {
-                                             let braid_lock = braid.read().await;
-                                             braid_lock.check_genesis_beads(&genesis)
-                                         };
-                                         match status {
-                                             braid::GenesisCheckStatus::GenesisBeadsValid => {
-                                                 info!(count = %genesis.len(), "Genesis beads validated");
-                                             }
-                                             braid::GenesisCheckStatus::MissingGenesisBead => {
-                                                 warn!(peer = %peer, "Missing genesis bead");
-                                             }
-                                             braid::GenesisCheckStatus::GenesisBeadsCountMismatch => {
-                                                 warn!(
-                                                     received = %genesis.len(),
-                                                     peer = %peer,
-                                                     "Genesis bead count mismatch"
-                                                 );
-                                             }
-                                         }
-                                     }
-                                     bead::BeadResponse::Error(error) => {
-                                         error!(error = ?error, "Bead sync response error");
-                                         peer_manager.update_score(&peer, -1.0);
-                                     }
-                                 };
-                             }
-                         }
-                     }
+                    request_response::Event::Message {
+                        peer,
+                        message,
+                        connection_id,
+                    },
+                )) => {
+                    info!(
+                        peer = %peer,
+                        message = ?message,
+                        connection = ?connection_id,
+                        "Bead sync message received"
+                    );
+                    match message {
+                        request_response::Message::Request {
+                            request,
+                            request_id: _,
+                            channel,
+                        } => {
+                            // Handle the bead sync request here
+                            match request {
+                                bead::BeadRequest::GetBeads(hashes) => {
+                                    let mut beads = Vec::new();
+                                    {
+                                        let braid_lock = braid.read().await;
```

Comment:
This is potentially painful for keeping in sync, since it blocks writers to the Braid (incoming beads) if it takes more than $a$ ~ 1s to complete this request, we will sit with beads in `incoming` while we send IBD blocks, which causes our mined beads to mine old parents, cohorts to become large, and our payouts to be decreased or lost. Most of the time, peers will be requesting historic blocks not near the tip, and we don't want to block writing the braid.

Instead, consider using the DB directly and not locking the braid here unless the request is very close to the tips.

We also need to develop a strategy for syncing the DB to disk. We're not doing it at all right now. Might not matter in operation though, we can still read it without locks...